app/controllers/concerns/atmosphere/api/application_controller_ext.rb

```diff
@@ -7,37 +7,80 @@ module ApplicationControllerExt
 
       included do
         before_action :set_project
-      end
 
-      def delegate_auth
-        current_user ? current_user.mi_ticket : nil
-      end
+        def delegate_auth
+          current_user ? current_user.mi_ticket : nil
+        end
 
-      def token_request?
-        mi_ticket || token
-      end
+        def token_request?
+          mi_ticket || jwt_token || token
+        end
 
-      private
+        def pdp
+          Rails.logger.debug("Matching PDP to request content...")
+          pdp = nil # TODO: use permit-nothing default pdp
 
-      def mi_ticket
-        params[Air.config.mi_authentication_key] ||
-          request.headers[Air.config.header_mi_authentication_key]
-      end
+          if
+          (
+            params[Air.config.mi_authentication_key] ||
+              request.headers[Air.config.header_mi_authentication_key]
+          )
+            Rails.logger.debug("MI ticket detected; using MI PDP")
+            pdp = MiApplianceTypePdp
+          elsif
+          (
+            params[Devise::Strategies::TokenAuthenticatable.key].presence ||
+              request.headers[Devise::Strategies::TokenAuthenticatable.header_key].presence
+          )
+            Rails.logger.debug("Private token detected; using default PDP")
+            pdp = Atmosphere::DefaultPdp
+          elsif
+          (
+            request.env['HTTP_AUTHORIZATION'] &&
+              request.env['HTTP_AUTHORIZATION'].match(/^Bearer /)
+          )
+            Rails.logger.debug("JWT token detected; using local PDP")
+            pdp = Atmosphere::LocalPdp
+          else
+            # TODO: Add a permit-nothing class to handle malformed requests gracefully.
```

Comment:
This is actually a matter for discussion rather than a design decision. We can continue to use `DefaultPDP` in these cases -- this will preserve the existing behavior. My comment was triggered by the observation that `DefaultPDP` is quite permissive, which looks like a security hole to me. What is your opinion?